### DISCUSSION TITLE
Optimistic update - generalization?

### DISCUSSION THREAD

**Discussion by dhmoon91** [trpc/trpc#5746] [Q&A]
Hi- I'm using trpc+react-query combo in my app. 

Let's say I have a `update_following` mutation that gets triggered from various source of app through `follow button`.

eg: 
'Post pages' (posts retrieved by `getPosts()` query) -> Per each post I have a 'follow button' -> Clicking will trigger `update_following` mutation

'My following pages' (posts retrieved by `getFollowingPosts()` query) ->  Per each my follwing's post, I have a 'follow button' -> Clicking will trigger `update_following` mutation

I want to update 'followedByMe` field of each post object from various pages that flips the text to 'Follow' vs 'Following'.

I already have one built for 'Post pages`: 
```
  const updateUserFollowingMut = trpc.user.updateUserFollowing.useMutation({
    /** Below is using optimistic update for discover page to update UI right away -
     * If UI lag is too much lag when clicking follow button, we might build this out fully for bunch of places that we do
     * reference: https://stackoverflow.com/questions/76070846/trpc-optimistic-updates-on-nested-arrays
     */
    onMutate: async (opts) => {
      await utils.post.getPosts.cancel();
      utils.post.getPosts.setInfiniteData({ limit: 3 }, (oldData) => {
        if (!oldData) {
          return {
            pages: [],
            pageParams: [],
          };
        }
        return {
          ...oldData,
          pages: oldData.pages.map((page) => ({
            ...page,
            posts: page.posts.map((post) => {
              if (storeId && post.store && storeId === post.storeId) {
                return {
                  ...post,
                  store: {
                    ...post.store,
                    // @ts-ignore
                    followedByMe: !post.store.followedByMe,
                  },
                };
              }
              if (staffId && post.staff && staffId === post.staffId) {
                return {
                  ...post,
                  staff: {
                    ...post.staff,
                    followedByMe: !post.staff.followedByMe,
                  },
                };
              }
              return post;
            }),
          })),
        };
      });
    },
    // Part of optimistic update
    onSettled: () => {
      utils.post.getPosts.refetch();
    },
  });
```

**This is working fine but my question is do we have to do the same for ALL queries (eg; Do the same for `getFollowingPosts.setInfiniteData(...)`), and all the other queries that needs to update `followedByMe`)?**

**Or is there some generalization/key setting that can be used to update `followedByMe` across the whole app's queries?**

eg: 
```

await utils.setFields(('fieldKey') => {
 return {fieldKey: !fieldKey}
})

```

**Comment by shanksxz** [comment]
did u find any solution?

**Reply by dhmoon91** [reply]
no not yet. 
I just duplicated this optimistic update in bunch of places for now
